meta-mbl-distro/scripts/create-update-payload

```diff
@@ -97,128 +103,279 @@ def parse_args():
+
+    :param filename_var_map dict: Mapping of bitbake variable names to files.
+    :param staging dir Path: path to the dir where the files will be copied to.
+    """
+    for var, paths in filename_var_map.items():
+        if var == MBL_BOOT_ID:
+            dst_dir = staging_dir / var
+            dst_dir.mkdir(exist_ok=True)
+            copy_files_to_subdir(paths, dst_dir)
+        else:
+            dst_dir = staging_dir
+            copy_single_file_modifying_name(paths, var, dst_dir)
+
+def add_payload_format_version_to_tar_archive(tar):
+    buf = io.BytesIO(PAYLOAD_FORMAT_VERSION)
+    tar_info = tarfile.TarInfo(name="payload_format_version")
+    tar_info.size = len(buf.getvalue())
+    tar_info.mode = 0o444
+    tar_info.uid = 0
+    tar_info.gid = 0
+    tar_info.uname = "root"
+    tar_info.gname = "root"
+    tar_info.mtime = time.time()
+    tar.addfile(tar_info, buf)
+
+def create_tar_archive(tar_archive_name, dir_path, fname_var_map):
     """
-    Create the payload tar archive.
+    Create a tar archive.
 
     The tar archive is uncompressed but the files within it are compressed
-    using xz.
+    using gz.
 
     :param tar_archive_name str: Path to the tar archive.
     :param dir_path Path: Path to the directory of the files to archive.
     :param fname_var_map dict: Mapping of bitbake variable names to files.
     """
     files_for_tar = []
     for fname in dir_path.iterdir():
-        # Look up a dictionary key by value, so we can name the file with
-        # the bitbake variable name.
-        gzip_fname = "{}.gz".format(
-            list(fname_var_map.keys())[
-                list(fname_var_map.values()).index(fname.name)
-            ]
-        )
+        for var in fname_var_map:
+            if fname.is_dir():
+                gzip_fname = "{}.tar.gz".format(fname.name)
+            elif fname.name == var:
+                gzip_fname = "{}.tar.gz".format(var)
+                break
+
         gzip_path = dir_path / gzip_fname
         files_for_tar.append(gzip_path)
 
-        with fname.open(mode="rb") as f_obj:
-            with gzip.open(str(gzip_path), "wb") as gzip_arc:
-                shutil.copyfileobj(f_obj, gzip_arc)
+        with tarfile.open(str(gzip_path), mode="w:gz") as tar_file:
+            tar_file.add(str(fname), arcname=gzip_path.name)
 
     with tarfile.open(tar_archive_name, mode="w") as tar:
+        add_payload_format_version_to_tar_archive(tar)
         for arc_path in files_for_tar:
             tar.add(str(arc_path), arcname=arc_path.name)
 
 
+def _bootloader_one_with_kernel(bootloader_components, tinfoil):
+    return (
+        get_bitbake_conf_var("MBL_WKS_BOOTLOADER1_SKIP", tinfoil) == "1"
+        and "1" in bootloader_components
+    )
+
+
+def _kernel_with_bootloader_one(bootloader_components, tinfoil):
+    return get_bitbake_conf_var("MBL_WKS_BOOTLOADER1_SKIP", tinfoil) == "1"
+
+
+def get_payload_files_data(args, tinfoil):
```

Comment:
I think this function needs a docstring.

Reply:
addressed in 6a9f0c1923038a73e67474879819cb529a0df79a